kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableUpdater.java

```diff
@@ -0,0 +1,92 @@
+package com.wepay.kafka.connect.bigquery.utils;
+
+import com.google.cloud.bigquery.TableId;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig;
+import org.apache.kafka.common.config.ConfigException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.wepay.kafka.connect.bigquery.utils.TopicToTableResolver.getTopicToTableSingleMatch;
+
+public class TopicToTableUpdater {
+  public static String getTopicToDateset (
+          List<Map.Entry<Pattern, String>> patterns,
+          String topicName,
+          String valueProperty,
+          String patternProperty) {
+
+    String match = null;
+    for (Map.Entry<Pattern, String> pattern : patterns) {
+      Matcher patternMatcher = pattern.getKey().matcher(topicName);
+      if (patternMatcher.matches()) {
+        if (match != null) {
+          String secondMatch = pattern.getValue();
+          throw new ConfigException(
+                  "Value '" + topicName
+                          + "' for property '" + valueProperty
+                          + "' matches " + patternProperty
+                          + " regexes for both '" + match
+                          + "' and '" + secondMatch + "'"
+          );
+        }
+        match = pattern.getValue();
+      }
+    }
+    if (match == null) {
+      throw new ConfigException(
+              "Value '" + topicName
+                      + "' for property '" + valueProperty
+                      + "' failed to match any of the provided " + patternProperty
+                      + " regexes"
+      );
+    }
+
+    return match;
+  }
+
+  /**
+   * Return a Map detailing which BigQuery table each topic should write to.
+   *
+   * @param config Config that contains properties used to generate the map
+   * @return A Map associating Kafka topic names to BigQuery table names.
+   */
+  public static Map<String, TableId> updateTopicToTable(
```

Comment:
You probably don't need to return a map here since the passed in `topicToTable` map will be modified.